hudi-common/src/main/java/org/apache/hudi/common/util/queue/BoundedInMemoryQueue.java

```diff
@@ -204,6 +204,7 @@ private boolean expectMoreRecords() {
    * singleton iterator for this queue.
    */
   private Option<O> readNextRecord() {
+    checkIfInterrupted();
```

Comment:
do we need to check if interrupted for every record processing? will there be any negative impact on perf due to this. This is all in memory, and should not be an issue. but wanted to see if this is absolutely necessary.

Reply:
Agree with you, I missed it. Let me think about if there is any other way to notify consumer so that they can stop asap when queue received a termination signal.